src/template_items/templateItem.controler.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { TemplateItemController } from "./templateItem.controller";
+import { TemplateItemService } from "./templateItem.service";
+import { mockTemplateItem } from "./templateItem.testData";
+import { GetAllTemplatesDto } from "./dto/getAllTemplate.dto";
+import { TemplateItemDto } from "./dto/template.dto";
+import { UpdateTemplateDto } from "./dto/updateTemplate.dto";
+
+describe("TemplateItemController", () => {
+  let controller: TemplateItemController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [TemplateItemController],
+      providers: [
+        {
+          provide: TemplateItemService,
+          useValue: {
+            getAllTemplates: jest
+              .fn()
+              .mockImplementation(() => Promise.resolve([mockTemplateItem])),
```

Comment:
![image](https://user-images.githubusercontent.com/76163407/195990973-d4040b4c-2e69-4071-acb6-517c04d1124b.png)
![image](https://user-images.githubusercontent.com/76163407/195991035-22b135bf-f173-463f-b6f6-885e6a67f703.png)

I don't know if the test should be written like this, the current test I think is a bit simple